[WIP] nickserv: add custom trigger phrase support

dgw opened the pull request:
Added commands:

* `SetTrigger <string>`: sets `<string>` as custom trigger phrase
* `SetTrigger`: (no arguments) clears custom trigger phrase
* `ViewTrigger`: shows current custom trigger phrase

If merged, will resolve #1611.

I'm opening this now to get feedback, but I'm not done with it. In particular, I haven't tested all edge cases yet (what happens when the custom trigger phrase is `""`, for example), but I'm out of time to work on this tonight. Feedback on what I have so far will help me decide whether to spend more time on this later.

psychon commented on modules/nickserv.cpp:
```diff
@@ -113,7 +133,9 @@ class CNickServ : public CModule {
-                 "type /msg NickServ IDENTIFY password") != CString::npos) &&
+                 "type /msg NickServ IDENTIFY password") != CString::npos ||
+             sMessage.StripControls_n().find(
+                 GetNV("CustomTrigger")) != CString::npos) &&
```
What if `GetNV("CustomTrigger")` is an empty string? This would always match, wouldn't it?

dgw replied:
Actually, I just made the if statement even more ridiculous to fix this. :joy_cat: See cc523d4…

dgw commented on modules/nickserv.cpp:
```diff
@@ -113,7 +133,10 @@ class CNickServ : public CModule {
-                 "type /msg NickServ IDENTIFY password") != CString::npos) &&
+                 "type /msg NickServ IDENTIFY password") != CString::npos ||
+             (!GetNV("CustomTrigger").empty() &&
+              sMessage.StripControls_n().find(
```
The second one is best, I'd agree. I'll update the PR shortly.

dgw commented:
Updated to incorporate feedback from @ilovelinux. Control codes are now removed from custom trigger text before it is saved.

When the time comes to squash this, let me know. The little tweaks I've made since opening this PR don't need their own commits, really—especially the `fixup` commit…

ilovelinux commented:
It doesn't seem to work with some phrases as trigger. I'll try to debug it on these days

ilovelinux commented:
I no longer can test this. Can someone do tests? It's a very good feature to implement
